docs/designs/conceptual-versioning.md

```diff
@@ -0,0 +1,474 @@
+# Versioning Dev Spec
+
+This document specifies Docfx vnext versioning dev design.
+
+## Conceptual versioning - Phase 1
+
+### 1 Scope
+
+#### 1.1 In scope
+
+##### For content writer
+
+1. Support expressing version constrains as [moniker range](#21-moniker-range)
+
+2. Support moniker zone markdown syntax.
+
+3. Support publishing multiple files with same *sitePath* but different monikerRange(mutually exclusive with others), where the appropriate file is selected by URL with query `view={moniker}`.
+
+4. All files will be built in one round, which will bring a better performance.
+
+##### For content reader
+
+1. Support viewing page with specific version by query `?view={moniker}`
+
+2. Support showing Toc with different version.
+
+3. [Empty page](#42-empty-page) should be handled.
+
+#### 1.2 Out of scope
+
+1. Not support specific version reference, including link and xref.
+
+    > If user reference another file by link/Uid with different monikerRange, the resolve result of this reference can not guaranteed.
+
+    Sample:
+
+    ```txt
+    |- articles/
+    |    |- folder1/
+    |    |   |- a.md
+    |    |- folder2/
+    |    |   |- b.md
+    |    |- folder3/
+    |        |- b.md
+    |- config.yml
+    ```
+
+    the content of `config.yml` is
+    ```yml
+    name: dotnet
+    content: "articles/**.{md,yml}"
+    monikerRange:
+        "articles/folder1/**.{md,yml}": "netcore-1.0"
+        "articles/folder2/**.{md,yml}": "netcore-1.0"
+        "articles/folder3/**.{md,yml}": "netcore-2.0"
+    routing:
+        "articles/folder1/": "articles/"
+        "articles/folder2/": "articles/"
+        "articles/folder3/": "articles/"
+    monikerDefinition: "https://api.docs.com/monikers/"
+    ```
+
+    If content writer want to reference `folder3/b.md` in `folder1/a.md` by link `'[B](../folder3/b.md)'`, when the reader click the link [B]() on page `{host}/{docset-base-path}/articles/a?view=netcore-1.0`, they will jump to page `{host}/{docset-base-path}/articles/b?view=netcore-1.0`, which is the output of file `folder2/b.md`.
```

Comment:
can writer using `[B](../folder3/b.md)?view=netcore-2.0` to resolve this problem?

Reply:
yes! but the final view page is not guaranteed!

---
In reply to: [229263766](https://github.com/dotnet/docfx/pull/3522#discussion_r229263766) [](ancestors = 229263766)